### PR TITLE
asusctl: 4.5.8 -> 4.6.0

### PR DESCRIPTION
**Pull request by colemickens**
###### Description of changes

Update `asusctl`/`rog-control-center` to latest.

WIP because there's a number of changes here, including switching the aura config to RON and I'm not sure I have all of the changes in the derivation yet.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by colemickens** [comment]
I don't really understand the logic behind vendoring the Cargo.locks instead of using them from src, and afaict the manual doesn't speak to this.

**Comment by colemickens** [comment]
And is there a reason we can't set appropriate variables for the Makefile and use it to install files instead of duplicating that information?

**Comment by colemickens** [comment]
cc: @k900 @aacebedo

File: pkgs/applications/system/asusctl/default.nix [hunk 1]
```diff
@@ -13,32 +13,30 @@
...
     owner = "asus-linux";
     repo = "asusctl";
     rev = version;
-    hash = "sha256-6AitRpyLIq5by9/rXdIC8AChMVKZmR1Eo5GTo+DtGhc=";
+    hash = "sha256-QXRUi/CiJTG04Kv/p3PWVloPIguLyfFQoiY6tCQIF/M=";
   };
 
   cargoLock = {
-    lockFile = ./Cargo.lock;
+    lockFile = "${src}/Cargo.lock";
```
Comment (K900) [inline pkgs/applications/system/asusctl/default.nix:26]: That's IFD.

Comment (colemickens) [reply pkgs/applications/system/asusctl/default.nix:26]: Ah, okay, I'll revert that and update the vendored lockfile then.

**Comment by colemickens** [comment]
As far as I can tell, this works. But it currently has an extra patch to be able to use `buildRustPackage` and then call `make install-data` with `prefix=$out`.

This enables us to not replicate all of the data install commands in the derivation.

I'm going to wait and see if upstream accepts the patch.

**Comment by colemickens** [comment]
Patch was merged upstream. This is tested on my GA402RJ, and confirmed that fan control + keyboard backlight controls are present.

Note, my model does not have an "Anime" display, so I can't validate if that's exactly patched correctly.

**Comment by colemickens** [comment]
Well, this shouldn't be merged yet, I didn't realize how comprehensive the `asusd` nixos module is, it probably needs to be updated as well.

**Comment by K900** [comment]
Superseded by #228951.

**Pull request closed without merging**